rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java

```diff
@@ -2004,6 +2013,57 @@ void should_SetTransferFlagToTrue_When_TransferringWithoutRequestBody() {
       assertThat(response.getBody().getWorkbasketSummary().getWorkbasketId())
           .isEqualTo("WBI:100000000000000000000000000000000006");
       assertThat(response.getBody().isTransferred()).isTrue();
+      assertThat(response.getBody().getOwner()).isNull();
+    }
+
+    @TestFactory
+    Stream<DynamicTest>
+        should_SetTransferFlagAndOwnerDependentOnRequestBody_When_TransferringTasks() {
+
+      Iterator<Pair<Boolean, String>> iterator =
+          Arrays.asList(Pair.of(true, "user-1-1"), Pair.of(false, "user-1-2")).iterator();
+      String url =
+          restHelper.toUrl(
+              RestEndpoints.URL_TRANSFER_WORKBASKET_ID, "WBI:100000000000000000000000000000000006");
+
+      List<String> taskIds =
+          Arrays.asList(
+              "TKI:000000000000000000000000000000000003",
+              "TKI:000000000000000000000000000000000004",
+              "TKI:000000000000000000000000000000000039");
+
+      ThrowingConsumer<Pair<Boolean, String>> test =
+          pair -> {
+            HttpEntity<Object> auth =
+                new HttpEntity<>(
+                    new TransferTaskRepresentationModel(pair.getLeft(), pair.getRight(), taskIds),
+                    RestHelper.generateHeadersForUser("admin"));
+            ResponseEntity<Map<String, Object>> response =
+                TEMPLATE.exchange(url, HttpMethod.POST, auth, ERROR_CODE);
+
+            assertThat(response.getBody()).isNotNull();
+            assertThat((List<String>) response.getBody().get("successfullyTransferredTaskIds"))
+                .hasSize(2);
+            Map<String, LinkedHashMap> failedTasks =
```

Comment:
Why don't you use the type "BulkResultTaskRepresentation" here?

Reply:
Yes, I have changed it